### ISSUE TITLE
Build failure: cacche (aarch64-darwin)

### ISSUE THREAD

**Issue by n8henrie** [NixOS/nixpkgs#316520]
### Steps To Reproduce

Build failure after https://github.com/NixOS/nixpkgs/pull/315714

```
$ nix build github:NixOS/nixpkgs#ccache
error: hash mismatch in fixed-output derivation '/nix/store/slqin5zn20zdhsciqw33bp7g4yhk27m0-source.drv':
         specified: sha256-0T9iJXnDX8LffhB/5hsfBNyZQ211f0lL/7dvTrjmiE0=
            got:    sha256-YHSr2pnk17QEdrIHInXX2eBFN9OGjdleaB41VLaqlnA=
error: 1 dependencies of derivation '/nix/store/gp3rhhflvqj9gsiwnpj1r46zxq0zy0sr-ccache-4.10.drv' failed to build
```

### Build log

```
these 2 derivations will be built:
  /nix/store/gpb62wdf1kp9irnpzfdkhv30xskj7rgf-source.drv
  /nix/store/gp3rhhflvqj9gsiwnpj1r46zxq0zy0sr-ccache-4.10.drv
building '/nix/store/slqin5zn20zdhsciqw33bp7g4yhk27m0-source.drv'...
source> trying https://github.com/ccache/ccache/archive/refs/tags/v4.10.tar.gz
source>   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
source>                                  Dload  Upload   Total   Spent    Left  Speed
source>   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
source> 100  564k  100  564k    0     0   449k      0  0:00:01  0:00:01 --:--:--  646k
source> unpacking source archive /private/tmp/nix-build-source.drv-0/v4.10.tar.gz
error: hash mismatch in fixed-output derivation '/nix/store/slqin5zn20zdhsciqw33bp7g4yhk27m0-source.drv':
         specified: sha256-0T9iJXnDX8LffhB/5hsfBNyZQ211f0lL/7dvTrjmiE0=
            got:    sha256-YHSr2pnk17QEdrIHInXX2eBFN9OGjdleaB41VLaqlnA=
error: 1 dependencies of derivation '/nix/store/gp3rhhflvqj9gsiwnpj1r46zxq0zy0sr-ccache-4.10.drv' failed to build
```

### Additional context

Hoping this isn't another darwin "case sensitivity" issue.

### Notify maintainers

From the PR:
@Nanotwerp
@kira-bruneau 

From ccache:

@kira-bruneau
@r-burns

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-darwin"`
 - host os: `Darwin 23.5.0, macOS 14.5`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.19.4`
 - channels(n8henrie): `""`
 - channels(root): `""`
 - nixpkgs: `/nix/store/x44yrfrlx00rsbab7lqc5f71qwjazb56-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by Nanotwerp** [comment]
Should be fixed by #316524. The issue also happens on `x86-64-linux`. I have no idea why the hash changed. I did update and test build the update about 30 minutes after `ccache` itself was updated, so maybe they backported a change into the `4.10` release or something?

**Comment by n8henrie** [comment]
Awesome, thanks. I assumed it was passing on linux and only failing on darwin due to some similar recent issues with case-sensitivity, had I known it was just a hash update I would have just fixed! Guess I should have checked.

Thanks!
